### PR TITLE
Elements that need to respond to toggling have no way of doing so

### PR DESCRIPTION
**Pull request by onyxrev**
I've added event dispatching from the Toggle component that fires events to which other components and libraries can respond. I'm using this with the text autosizing component to autosize textareas that reside inside containers that initially start hidden and become visible by way of toggling.

### CODE REVIEW

File: app/assets/javascripts/megatron/utils/text-helpers.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ var TextHelpers = {
     }
     Array.prototype.forEach.call(document.querySelectorAll('textarea:not(.no-auto-size)'), autoHeight)
 
-    bean.on(document.querySelector('body'), 'keyup', 'textarea', function(event){
+    bean.on(document.querySelector('body'), 'keyup, toggler:show', 'textarea', function(event){
```
Comment (mbyczkowski) [inline app/assets/javascripts/megatron/utils/text-helpers.js]: @onyxrev is `'keyup, toggler:show'` intentional or is this missing couple single-quotes?

Comment (onyxrev) [inline app/assets/javascripts/megatron/utils/text-helpers.js]: Good question. Looks like it actually shouldn't have the comma:

Example:
`bean.on(element, 'keydown keyup', handler);`

See [https://www.npmjs.com/package/bean](https://www.npmjs.com/package/bean)

It's working, though, at least for the toggler event. The keyup event may be broken with the errant comma.

Comment (mbyczkowski) [inline app/assets/javascripts/megatron/utils/text-helpers.js]: cool, let's fix that then and also version bump the gem like here: https://github.com/compose-ui/megatron.rb/commit/639dd8295cfec26d63448936687ba921c278e251

**Comment by jeromegn** [comment]
Looks like there are merge conflicts.

LGTM once those conflicts are resolved :)

**Comment by onyxrev** [comment]
I believe this work has been integrated elsewhere. https://github.com/compose-ui/megatron.rb/commit/994f86f591ec22a59362df15b6de354315307048

**Pull request closed without merging**